packages/grafana-flamegraph/src/FlameGraph/FlameGraphContextMenu.tsx

```diff
@@ -1,12 +1,19 @@
 import React from 'react';
 
-import { MenuItem, MenuGroup, ContextMenu } from '@grafana/ui';
+import { MenuItem, MenuGroup, ContextMenu, IconName } from '@grafana/ui';
 
 import { ClickedItemData } from '../types';
 
-import { CollapseConfig } from './dataTransform';
+import { CollapseConfig, FlameGraphDataContainer } from './dataTransform';
+
+export type ExtraContextMenuButton = {
+  label: string;
+  icon: IconName;
+  onClick: (clickedItemData: ClickedItemData, data: FlameGraphDataContainer) => void;
```

Comment:
I think we should pass the actuall dataFrame here instead. The `FlameGraphDataContainer` is just implementation detail to help with rendering and honestly thought about refactoring it so I don't want to expose it to the outside. I also don't think there is much that would help for this in specific. The tree structure that it provides may be interesting, but again that changed a lot depending on the rendering algorithm so would like to also keep it private if we don't need to pass it down.

Reply:
Dima is out now,  @aocenas do you think you could move this forward. Sounds like this was enough but I'll integrate with whatever changes you think is best for this feature.

Reply:
The reason why we needed this is to be able to fetch all the parent from the click point like this
```typescript
let node: LevelItem | undefined = data.item;
    let labels = [];

    while (node) {
      for (const idx of node.itemIndexes) {
        labels.push(container.getLabel(idx));
      }

      node = node.parents?.[0];
    }

    labels = labels.filter((s) => s.localeCompare('total'));
    labels = labels.reverse();
```

May be you have a better idea ?

Reply:
It's actually not a problem because this feature won't be in the sandwich view but however we would need to know the state of the flamegraph to not show the button if the flamegraph is in sandwich view.